contracts/PoolManager.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: HORNEY
+
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./Authorizable.sol";
+import "./interfaces/IRewardPool.sol";
+
+contract PoolManager is Authorizable {
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using SafeERC20 for IERC20;
+
+    EnumerableSet.AddressSet private fullPoolList;
+    EnumerableSet.AddressSet private legacyPoolList;
+    IERC20 public governanceToken;
+
+    event AddPools(address[] pools, bool isLegacy);
+    event AddPool(address pool, bool isLegacy);
+    event RemovePools(address[] pools, bool isLegacy);
+    event GovernanceTokenChange(IERC20 newGovernanceToken);
+
+    constructor(IERC20 _governanceToken) {
+        governanceToken = _governanceToken;
+    }
+
+    /*
+        Write Functions - Permissioned
+    */
+
+    function addPool(address _pool, bool _isLegacy) public onlyAuthorized {
+        if (_isLegacy) {
+            if (!legacyPoolList.contains(_pool)) {
+                legacyPoolList.add(_pool);
+            }
+        } else {
+            if (!fullPoolList.contains(_pool)) {
+                fullPoolList.add(_pool);
+            }
+        }
+
+        emit AddPool(_pool, _isLegacy);
+    }
+
+    function addPools(address[] calldata _pools, bool _isLegacy) public onlyAuthorized {
+        if (_isLegacy) {
+            for (uint i = 0; i < _pools.length; i++) { 
+                if (!legacyPoolList.contains(_pools[i])) {
+                    legacyPoolList.add(_pools[i]);
+                }
+            }
+        } else {
+            for (uint i = 0; i < _pools.length; i++) { 
+                if (!fullPoolList.contains(_pools[i])) {
+                    fullPoolList.add(_pools[i]);
+                }
+            }
+        }
+
+        emit AddPools(_pools, _isLegacy);
+    }
```

Comment:
Only because I am jealous of Tastics review and also want you to become a Solidity super hero. 

Could this be simplified like this?

```solidity
function addPools(address[] calldata _pools, bool _isLegacy) public onlyAuthorized {
            for (uint i = 0; i < _pools.length; i++) { 
                addPool(_pools[i], _isLegacy);
            }

        emit AddPools(_pools, _isLegacy); // Event is optional now
    }
```

Reply:
Love it, done! 🍌